WordPress/Jetpack/Classes/System/JetpackWindowManager.swift

```diff
@@ -5,25 +5,54 @@ class JetpackWindowManager: WindowManager {
     /// receives migration flow updates in order to dismiss it when needed.
     private var cancellable: AnyCancellable?
 
+    var shouldImportMigrationData: Bool {
+        return !AccountHelper.isLoggedIn && !UserPersistentStoreFactory.instance().isJPContentImportComplete
+    }
+
     override func showUI(for blog: Blog?) {
         // If the user is logged in and has blogs sync'd to their account
         if AccountHelper.isLoggedIn && AccountHelper.hasBlogs {
-            shouldShowMigrationUI ? showMigrationUI(blog) : showAppUI(for: blog)
+            showAppUI(for: blog)
             return
         }
 
-        // Show the sign in UI if the user isn't logged in
         guard AccountHelper.isLoggedIn else {
-            showSignInUI()
+            if shouldImportMigrationData {
+                importAndShowMigrationContent(blog) { [weak self] in
+                    self?.showSignInUI()
+                }
+            } else {
+                showAppUI(for: blog)
```

Comment:
@alpavanoglu Maybe this line should be `self.showSignInUI()`, because `AcountHelper.isLoggedIn` is `false`.

Reply:
Makes sense. Simpler to understand that way. Fixing...